### ISSUE TITLE
Improvements Based on Feedback from 2/2/23

### ISSUE THREAD

**Issue by venuswku** [venuswku/data-visualizer#8]
1. Explore ways to decrease loading time for creating time-series.
     - [x] using [KD Trees](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KDTree.html) to get data points near a transect (2/13/23 & 2/14/23)
     - [x] using asyncio to asynchronously extract data along a transect (2/15/23 & 2/21/23)
          - [threading](https://panel.holoviz.org/user_guide/Performance_and_Debugging.html#threading) can help with computations in Panel
          - [concurrent futures](https://docs.python.org/3/library/concurrent.futures.html)
          - [semaphores](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Semaphore) are used in Seg2Map to limit the number of tasks that can be run concurrently without crashing the CPU
2. Display time-series data one at a time on the map when a file gets selected.
     - [x] rename "Dataset" widget to "Collection" (2/6/23)
     - [x] group data files by type of data - topography, bathymetry, DEM, grainsize, etc. (2/8/23 & 2/9/23)
     - [x] refactor code to only display the most recently selected data file (2/7/23)
     - [x] add info buttons in case user needs help with figuring out how to use the sidebar controls (2/8/23)
     - [x] display error message when incompatible data files - e.g. uses different measurements for the time-series' y-axis values - are chosen (2/10/23)
     - [x] experiment with using DataShader & compare loading times (2/14/23)
3. Don't let user choose data files that have mismatching y-axis measurements in the time-series.
     - [x] use `F-W Mean` for grainsize data's time-series (2/7/23)
          - ask Jon and Andrew: Do adding error bars for grainsize data's time-series make it more useful?
          - Dan suggested to use `F-W Sorting` for error bars (add and subtract from `F-W Mean` for error bar's range)
